examples/Lorenz/GModel_common.jl

```diff
@@ -0,0 +1,358 @@
+
+using DocStringExtensions
+
+using Random
+using Distributions
+using LinearAlgebra
+using FFTW
+
+"""
+GSettings{FT<:AbstractFloat, KT, D}
+
+Structure to hold all information to run the forward model G
+
+# Fields
+$(DocStringExtensions.FIELDS)
+"""
+struct GSettings{FT <: AbstractFloat, KT, D}
+    "model output"
+    out::Array{FT, 1}
+    "time period over which to run the model, e.g., `(0, 1)`"
+    tspan::Tuple{FT, FT}
+    "x domain"
+    x::Array{FT, 1}
+    "y domain"
+    y::Array{FT, 1}
+end
+
+struct LSettings
+    # Stationary or transient dynamics
+    dynamics::Int32
+    # G model statistics type
+    stats_type::Int32
+    # Integration time start
+    t_start::Float64
+    # Data collection length
+    T::Float64
+    # Integration length
+    Ts::Float64
+    # Duration of polynomial fit, number of samples of Ts to fit over
+    Tfit::Float64
+    # Initial perturbation
+    Fp::Array{Float64}
+    # Number of longitude steps
+    N::Int32
+    # Timestep
+    dt::Float64
+    # Simulation end time
+    tend::Float64
+    # For stats_type=2, number of frequencies to consider
+    kmax::Int32
+    # Should ω be learned using CES?
+    ω_fixed::Bool
+    # Truth ω
+    ω_true::Float64
+end
+
+struct LParams
+    # Mean forcing
+    F::Float64
+    # Forcing frequency for transient terms
+    ω::Float64
+    # Forcing amplitude for transient terms
+    A::Float64
+end
+
+"""
+    run_G_ensemble(params::Array{FT, 2},
+                   settings::GSettings{FT},
+                   update_params,
+                   moment,
+                   get_src;
+                   rng_seed=42) where {FT<:AbstractFloat}
+
+Run the forward model G for an array of parameters by iteratively
+calling run_G for each of the N_ensemble parameter values.
+Return g_ens, an array of size N_data x N_ensemble, where
+g_ens[:,j] = G(params[:,j])
```

Comment:
This isn't exactly correct in the current implementation, but I'm happy to leave this for a future PR that cleans up the runscript